server/app/services/export/QuestionJsonPresenter.java

```diff
@@ -0,0 +1,275 @@
+
+      if (question.getDateValue().isPresent()) {
+        LocalDate date = question.getDateValue().get();
+        return ImmutableMap.of(path, DateTimeFormatter.ISO_DATE.format(date));
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+  }
+
+  class EmptyJsonPresenter implements QuestionJsonPresenter<Question> {
+    @Override
+    public ImmutableMap<Path, ?> getJsonEntries(Question question) {
+      return ImmutableMap.of();
+    }
+  }
+
+  class NumberJsonPresenter implements QuestionJsonPresenter<NumberQuestion> {
+    @Override
+    public ImmutableMap<Path, Long> getJsonEntries(NumberQuestion question) {
+      Path path = question.getNumberPath().asApplicationPath();
+
+      if (question.getNumberValue().isPresent()) {
+        return ImmutableMap.of(path, question.getNumberValue().get());
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+  }
+
+  class PhoneJsonPresenter implements QuestionJsonPresenter<PhoneQuestion> {
+
+    private static final PhoneNumberUtil PHONE_NUMBER_UTIL = PhoneNumberUtil.getInstance();
+
+    @Override
+    public ImmutableMap<Path, String> getJsonEntries(PhoneQuestion question) {
+      Path path = question.getPhoneNumberPath().asApplicationPath();
+
+      if (question.getPhoneNumberValue().isPresent()
+          && question.getCountryCodeValue().isPresent()) {
+        String formattedPhone =
+            getFormattedPhoneNumber(
+                question.getPhoneNumberValue().get(), question.getCountryCodeValue().get());
+        return ImmutableMap.of(path, formattedPhone);
+      } else {
+        return ImmutableMap.of();
+      }
+    }
+
+    /**
+     * This method accepts a phoneNumber as String and the countryCode which is iso alpha 2 format
+     * as a String. It formats the phone number per E164 format. For a sample input of
+     * phoneNumberValue="2123456789" with countryCode="US", the output will be +12123456789
+     */
+    private static String getFormattedPhoneNumber(String phoneNumberValue, String countryCode) {
+      try {
+        Phonenumber.PhoneNumber phoneNumber =
+            PHONE_NUMBER_UTIL.parse(phoneNumberValue, countryCode);
+        return PHONE_NUMBER_UTIL.format(phoneNumber, PhoneNumberUtil.PhoneNumberFormat.E164);
+      } catch (NumberParseException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  class SingleSelectJsonPresenter implements QuestionJsonPresenter<SingleSelectQuestion> {
+    @Override
+    public ImmutableMap<Path, String> getJsonEntries(SingleSelectQuestion question) {
+      return ImmutableMap.of(
+          question.getApplicantQuestion().getContextualizedPath().join(Scalar.SELECTION),
+          question
+              .getApplicantQuestion()
+              .createSingleSelectQuestion()
+              .getSelectedOptionValue(LocalizedStrings.DEFAULT_LOCALE)
+              .map(LocalizedQuestionOption::optionText)
+              .orElse(""));
+    }
+  }
```

Comment:
Above code is taken from `case DROPDOWN`:

https://github.com/civiform/civiform/blob/358ad3dbb6e67aacb91afb3ede60e32ac281ba26/server/app/services/applicant/ReadOnlyApplicantProgramServiceImpl.java#L450-L508